ee/server/service/teams.go

```diff
@@ -586,23 +586,34 @@ func (svc *Service) DeleteTeam(ctx context.Context, teamID uint) error {
 }
 
 func (svc *Service) GetTeam(ctx context.Context, teamID uint) (*fleet.Team, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.Team{ID: teamID}, fleet.ActionRead); err != nil {
-		return nil, err
+	alreadyAuthd := svc.authz.IsAuthenticatedWith(ctx, authz_ctx.AuthnDeviceToken)
+	if !alreadyAuthd {
+		if err := svc.authz.Authorize(ctx, &fleet.Team{ID: teamID}, fleet.ActionRead); err != nil {
+			return nil, err
+		}
```

Comment:
should we check that the authorized entity belongs to the team? (so device tokens can't read other teams?)

Reply:
Good call, I'll add it in Dante's PR that I'm finishing up if you don't mind (I'll merge this one as-is). Will just be less branch switching.

Reply:
sounds good!